packages/components/src/mobile/audio-player/audio-url-parser.native.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * WordPress dependencies
+ */
+import { safeDecodeURI } from '@wordpress/url';
+
+export const parseAudioUrl = ( src ) => {
+	const decodedURI = safeDecodeURI( src );
+	const fileName = decodedURI
+		.split( '#' )
+		.shift()
+		.split( '?' )
+		.shift()
+		.split( '/' )
+		.pop();
+	const parts = fileName.split( '.' );
+	const extension = parts.length === 2 ? parts.pop().toUpperCase() + ' ' : '';
```

Comment:
I see that if the file contains an extension. we're adding an extra blank space at the end of the string, what's the reason for this?

Reply:
Ah, yes, I decided to add a space here because there are times when the `extension` isn't present but if it is I need to add the space here so within the UI itself the extension doesn't get attached to the `audio file` Text in the UI. For more context, see this comment https://github.com/WordPress/gutenberg/pull/27817#discussion_r616350899

Reply:
Oh ok, I'm wondering if the extra space should be handled then in the component instead since it's specific to how to render it, wdyt? My first impression was that this method was agnostic to the component but maybe since it's located under the same folder makes sense to include it 🤔 .

Reply:
You are right about the component handling this especially since it's UI-related. I could create another small PR to resolve this since this PR as solved the main need which was to add the tests 😄